public/components/data_source_top_nav_menu.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo, useContext, useEffect, useCallback } from 'react';
+
+import type { CoreStart, MountPoint, SavedObjectsStart } from '../../../../src/core/public';
+import type {
+  DataSourceManagementPluginSetup,
+  DataSourceSelectableConfig,
+} from '../../../../src/plugins/data_source_management/public';
+import type { DataSourcePluginSetup } from '../../../../src/plugins/data_source/public';
+import { DataSourceContext } from '../contexts/data_source_context';
+
+export interface DataSourceTopNavMenuProps {
+  notifications: CoreStart['notifications'];
+  savedObjects: SavedObjectsStart;
+  dataSource?: DataSourcePluginSetup;
+  dataSourceManagement?: DataSourceManagementPluginSetup;
+  setActionMenu: (menuMount: MountPoint | undefined) => void;
+}
+
+export const DataSourceTopNavMenu = ({
+  dataSource,
+  savedObjects,
+  notifications,
+  setActionMenu,
+  dataSourceManagement,
+}: DataSourceTopNavMenuProps) => {
+  const dataSourceEnabled = !!dataSource?.dataSourceEnabled;
```

Comment:
This component is only mounted when the dataSourceEnabled flag is true, do we need to double check the flag?

Reply:
Do we mean the `dataSourceEnabled` will always be true, if the top nav menu mounted? Add check here is for `dataSourceEnabled` changed. The `dataSourceEnabled` in DataSourceContextProvider will be updated if enabled status changed from outside.